woocommerce/woocommerce-ios
# CurrencySettings: it no more depends on `ResultController` and `ServiceLocator`

## Description
As discussed here p91TBi-3bP since it will come in handy in the future, it would be good to not expose `ResultController` and `ServiceLocator` to the `CurrencySettings` class, that will be useful in the future for using it in a Widget extension. In that way, we don't need to expose to a potential widget these two big dependencies.
I created a `SelectedSiteSettings` class where I moved all the logic about fetching the latest Site Settings, that takes care also to update the Currency Settings used in the app, and in the future will be helpful also for other things (for example, for selecting the Site Country. I've in the plan to open a separate PR for it).

## Testing
- CI should pass
- **Sanity Check:**
1. Open a store with a specific currency, and navigate to the Order List.
2. The prices of the orders should be correct like in `develop`.
3. Switch to another store that uses a different currency.
4. Navigate to the Order List -> Prices should be updated using the new currency value.

Update release notes:
